staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go

```diff
@@ -77,17 +95,42 @@ func (c *APIServiceRegistrationController) sync(key string) error {
 		return err
 	}
 
-	c.apiHandlerManager.AddAPIService(apiServer)
+	c.apiHandlerManager.AddAPIService(apiService, c.getDestinationHost(apiService))
 	return nil
 }
 
+func (c *APIServiceRegistrationController) getDestinationHost(apiService *apiregistration.APIService) string {
+	if apiService.Spec.Service == nil {
+		return ""
+	}
+
+	destinationHost := apiService.Spec.Service.Name + "." + apiService.Spec.Service.Namespace + ".svc"
+	service, err := c.serviceLister.Services(apiService.Spec.Service.Namespace).Get(apiService.Spec.Service.Name)
+	if err != nil {
+		return destinationHost
+	}
+	switch {
+	// use IP from a clusterIP
+	case service.Spec.Type == v1.ServiceTypeClusterIP:
+		return service.Spec.ClusterIP
+	}
```

Comment:
> if type is ExternalName, wouldn't it make sense to use the service.spec.ExternalName (possibly as the ServerName for TLS validation as well)

Maybe. I'm not even completely certain the option does what I think it does.  It appeared to in my manual test, but do you know for sure one way or the other?